app/controllers/concerns/idv/doc_auth_vendor_concern.rb

```diff
@@ -6,8 +6,16 @@ module DocAuthVendorConcern
 
     # @returns[String] String identifying the vendor to use for doc auth.
     def doc_auth_vendor
-      bucket = ab_test_bucket(:DOC_AUTH_VENDOR)
+      if resolved_authn_context_result.facial_match? && default_vendor_is_not_mock?
```

Comment:
what if a facial match is required and the default vendor is  :mock? seems like the user could be routed to a vendor that does not support facial match? 🤔

Reply:
yeah, say default is socure and facial match is requested. Maybe:
```
bucket = ab_test_bucket(:DOC_AUTH_VENDOR
if resolved_authn_context_result.facial_match? && bucket != :mock
  bucket = :lexis_nexus
end
```

Reply:
furthering Doug's point ... this appears to assume that  mock or lexis nexis is enabled as a doc auth vendor.  It seems that we'd need to know whether a vendor is enabled before routing to that vendor. what happens if only socure is enabled, would we have a doc auth vendor when a facial match is req'd 🤔

Reply:
> furthering Doug's point ... this appears to assume that mock or lexis nexis is enabled as a doc auth vendor. It seems that we'd need to know whether a vendor is enabled before routing to that vendor. what happens if only socure is enabled, would we have a doc auth vendor when a facial match is req'd 🤔

If LN is not enabled should we just redirect them to an error page?

Reply:
> If LN is not enabled should we just redirect them to an error page?

great question ... @tahineemay ? ☝🏿

Reply:
> > furthering Doug's point ... this appears to assume that mock or lexis nexis is enabled as a doc auth vendor. It seems that we'd need to know whether a vendor is enabled before routing to that vendor. what happens if only socure is enabled, would we have a doc auth vendor when a facial match is req'd 🤔
> 
> If LN is not enabled should we just redirect them to an error page?

i tested how that operates today and the standard document capture page is rendered but then submitting the document fails 

<img width="659" alt="Screenshot 2024-11-26 at 7 13 54 AM" src="https://github.com/user-attachments/assets/5f1ab06f-d36e-44f4-9195-6363c2548660">

Reply:
In this unlikely case, that a user requires facial match and only Socure is configured, we have to throw and error to help surface the misconfiguration. We cannot have the user go through a non-IAL2 IDV, if that is their requirement.

Reply:
as per @tahineemay response we should return the doc_auth_vendor should be nil  which render the current behavior when there is no vendor 👍🏿

Reply:
btw, when doc_auth_vendor is nil, this [ticketed bug](https://cm-jira.usa.gov/browse/LG-15200) is throwing a 500. for your testing it maybe helpful to comment out the redirect_to_correct_vendor when testing doc_auth_vendor is nil 👍🏿